app/models/user.rb

```diff
@@ -407,6 +413,22 @@ def max_changeset_comments_per_hour
     end
   end
 
+  def max_changes_per_hour
+    if moderator?
+      Settings.moderator_changes_per_hour
+    elsif importer?
+      Settings.importer_changes_per_hour
+    else
+      first_change = changesets.reorder(:created_at => :asc).pick(:created_at) || Time.now.utc
```

Comment:
would it make sense to have time since first edit, after expiration of last block (if user was ever blocked)

That could help with accounts that mass-imported stuff, got block/zero hour block and then were importing things again on large scale shortly after block expired

(assuming that it is really easy to get latest block expiration and it has no noticeable performance impact)